api/core/v2/entity.go

```diff
@@ -131,6 +131,14 @@ func FixtureEntity(name string) *Entity {
 	}
 }
 
+// FixtureProxyEntity returns a testing fixture for an proxy Entity object.
+func FixtureProxyEntity(name string) *Entity {
+	e := FixtureEntity(name)
+	e.EntityClass = "proxy"
+	e.ObjectMeta.Name = ""
```

Comment:
What's the rationale of zeroing out the proxy entity's name?

Reply:
The combination of `EntityClass` and `Entity.ObjectMeta.Name` are what allowed this change to differentiate between the three different types of events the agent forwards to the backend without having to add a new field to the event object:

1. Non-proxy checks (`EntityClass: "host"` and `Entity.ObjectMeta.Name: "foo-agent")
2. Proxy check results submitted via API (`EntityClass: "proxy"` and `Entity.ObjectMeta.Name: "")
3. Proxy checks scheduled by backend (`EntityClass: "proxy" and `EntityObjectMeta.Name: "<proxy-entity-name>"`)

The backend can then, after looking up the entity in the store based on `Event.Check.ProxyEntityName` as it previously already did, compare the entity in the store to the entity name in the submitted event. If there's no entity in the store, but `Event.Entity.ObjectMeta.Name` is non-empty, then the backend knows that the entity was deleted and the event should be dropped.

Perhaps this fixture method could be named better. If you look at the tests you'll see that they use either `FixtureEntity` or `FixtureProxyEntity` depending on which of the above cases they are emulating.